novuhq/novu
# [Onboarding]: NV-3415 | Feature flag and conditional rendering setup

### What change does this PR introduce?

- Refactors the top-level `App` to separate it from the `Routes` and non-React initialization steps
- Adds the feature flag for Onboarding and related hook setup
- Little high-level tweaks for `Get started` page
- Stylistic tweaks to the timeline + semantic HTML updates

### Why was this change needed?

- Refactoring was necessary to enable us to use hooks and feature flags to conditionally handle routes
  - Furthermore, this provides clearer boundaries for App organization in line with the Single Responsibility principle
- Solves [NV-3415](https://linear.app/novu/issue/NV-3415/🙋-onboarding-feature-flag)

### Other information

- There is now proper coloring of the bullets and timeline connector, along with line heights of the text paragraphs
<img width="1457" alt="Screenshot 2024-02-08 at 1 09 02 PM" src="https://github.com/novuhq/novu/assets/47441786/5f770759-df68-4074-b344-13af93fcf06f">- _The new onboarding flow will appear all the time for now because the default value of the hook is true_ 

### Validation steps

1. Check-out the branch and login
2. Navigate to the `Get started` page
3. The new (non-functional) flow should still appear
